test/message-parserSpec.ts

```diff
@@ -17,7 +17,7 @@ describe('message parser', () => {
       }
       it (`parses ${TOPIC[topic]} messages ${authAction} correctly`, () => {
         const result = parse(spec.urp.value)[0]
-        expect(result.parseError).to.be.undefined
+        expect(result.parseError).to.be.an('undefined')
```

Comment:
whats the difference with these out of curiosity?

Reply:
This one
```javascript
expect(result.parseError).to.be.undefined
```
hides information of an method call in an getter method called when you call an attribute. This might be fine, but on a reviewer perspective I'm not able to predict side effects or able to jump in the method, which is called by accessing the attribute. I'm also not so happy with my one, because it mixes types, which is also a problem. This one was provided as a solution on the documentation of chai in **bdd** style. 

On my opinion the best is to switch to the assert methods which are also provided by chai. The [isUndefined](http://chaijs.com/api/assert/#method_isundefined)-method should do what expected.